### PR TITLE
viz-v2 first commit

### PR DESCRIPTION
**Pull request by iris-qu**
My current working branch (-b iris) is too many commits behind master, so I'm starting a new branch with the following content:
- Static Vue.js project with Three.js visualization
- Express server that interacts with BigQuery API
-Wiki categorization code

dev link: [https://v2-dot-wikidetox-viz.appspot.com/](https://v2-dot-wikidetox-viz.appspot.com/)

TODO:
- Action buttons integration with wikipedia
- Some frontend refresh
- Mobile view
- Data streaming and wiki categorization integration

### CODE REVIEW

File: viz_v2/wiki_categorize/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (iislucas) [inline viz_v2/wiki_categorize/package.json]: Can you add a README to this directory to explain what this npm project is about?

File: viz_v2/wiki_categorize/src/bigQueryData.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        this.dataset = this.bigquery.dataset(this.datasetID);
+        this.table = this.dataset.table(this.distDataTable);
+    }
+
+    async querySourceTable() {
+
+        // max page ID 57726837
+        const sqlQuery =    `SELECT * 
+                                FROM \`${this.projectId}.${this.datasetID}.${this.originalDataTable}\`
+                            WHERE page_id >= 43295130`;
```
Comment (iislucas) [inline viz_v2/wiki_categorize/src/bigQueryData.ts]: where does this magic number come from? Add a comment to explain?

Comment (iris-qu) [reply viz_v2/wiki_categorize/src/bigQueryData.ts]: Ah I was experimenting with this. This is the talk page categorization code separate from the vis, I'll clean it up.

File: viz_v2/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+To run code using your local machine (not via the docker
+environment), you will need to install `nodejs` (e.g., by [installing NVM](https://github.com/creationix/nvm)).
+From project root, run
+
+```
+yarn install
+```
+to install dependencies and create a `yarn.lock` file.
+
+This project relies on Google's [BigQuery API](https://cloud.google.com/bigquery/docs/reference/rest/v2/) to query data, via an express server. To setup an instance you need a Google Cloud Project with the Perpsective API enabled.
```
[the remark this answers is not in the document]
Comment (iris-qu) [reply viz_v2/README.md]: Right now everything lives in a partitioned table that only has 2017 onward data. Considering the amount of work, we should probably leave it as it is for wiki workshop. 
Jeff is helping with data streaming -- if we want data to live on spanner we should flag this to him.

[the remark this answers is not in the document]
Comment (iislucas) [reply viz_v2/README.md]: SGTM

File: viz_v2/wiki_categorize/category_data/buildCategories.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const data = require('./Cloud_categories.json');
+const fs = require('fs');
+
+const categories = [];
```
[the remark this answers is not in the document]
Comment (iris-qu) [reply viz_v2/wiki_categorize/category_data/buildCategories.js]: I was using this code for an early experiment, the actual categorization process doesn't require it. I'll delete the folder.

**Comment by iris-qu** [comment]
Made changes to:
- Deleted natural language API categories files
- Added README to wiki_categorize

Let me if this looks good :)

File: viz_v2/wiki_categorize/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "ES2015",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
```
Comment (iislucas) [inline viz_v2/wiki_categorize/tsconfig.json]: Lots of commented out lines here too... you want them or remove them?

**Pull request merged**